beacon-chain/cache/depositsnapshot/leaf_node.go

```diff
@@ -0,0 +1,36 @@
+package depositsnapshot
+
+import "github.com/pkg/errors"
+
+// Leaf represents a leaf node holding a deposit and satisfies the MerkleTreeNode interface.
+type Leaf struct {
+	hash [32]byte
+}
+
+// GetRoot returns the root of the Merkle tree.
+func (l *Leaf) GetRoot() [32]byte {
+	return l.hash
+}
+
+// IsFull returns whether there is space left for deposits.
```

Comment:
Add to the commentary here why a leaf is always considered full.

Reply:
Additional commentary added, could you please check that it is clear.